account/lnurl.go

```diff
@@ -29,14 +30,60 @@ type LoginResponse struct {
 	Token string `json:"token"`
 }
 
-func (a *Service) HandleLNURL(rawString string) (*data.LNUrlResponse, error) {
+func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err error) {
+	var iparams lnurl.LNURLParams
+	var rawurl, lightningAddress string
+
 	encodedLnurl, ok := lnurl.FindLNURLInText(rawString)
-	if !ok {
-		return nil, fmt.Errorf("'%s' does not contain an LNURL.", rawString)
+	if ok {
+		a.log.Infof("HandleLNURL %v", encodedLnurl)
+		rawurl, iparams, err = lnurl.HandleLNURL(encodedLnurl)
+
+	} else {
+
+		err = fmt.Errorf("%q does not contain an LNURL.", rawString)
+
+		if strings.Contains(rawString, "@") {
+			// ref. Lightning-Address https://github.com/andrerfneves/lightning-address/blob/master/DIY.md
+
+			err = nil
+			parts := strings.Split(rawString, "@")
```

Comment:
Can we use the standard function https://pkg.go.dev/net/mail#ParseAddress instead

Reply:
Yes, to validate the address. But, as far as I can tell, it doesn't do the `username`, `host` splitting that we need to construct the lnurl.